File: 03-healthcare-cura/03-automation/src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage {
    private final WebDriver driver;
    private final By username = By.id("txt-username");
    private final By password = By.id("txt-password");
    private final By loginButton = By.id("btn-login");

    public LoginPage(WebDriver driver) {
        this.driver = driver;
    }

    public void open() {
        driver.get("https://katalon-demo-cura.herokuapp.com/profile.php#login");
    }

    public void login(String user, String pass) {
        if (!user.isEmpty()) driver.findElement(username).sendKeys(user);
        if (!pass.isEmpty()) driver.findElement(password).sendKeys(pass);
        driver.findElement(loginButton).click();
    }
}

File: 04-banking-parabank/03-automation/pages/login_page.py
from selenium.webdriver.common.by import By

BASE_URL = "https://parabank.parasoft.com/parabank/index.htm"


class LoginPage:
    def __init__(self, driver):
        self.driver = driver

    def open(self):
        self.driver.get(BASE_URL)

    def login(self, username, password):
        self.driver.find_element(By.NAME, "username").send_keys(username)
        self.driver.find_element(By.NAME, "password").send_keys(password)
        self.driver.find_element(By.CSS_SELECTOR, "input[value='Log In']").click()

    def get_error_text(self):
        return self.driver.find_element(By.CSS_SELECTOR, "#rightPanel .error").text
